soc: Add Ambiq Apollo4 Blue Plus SoC Support

aaronyegx opened the pull request:
This PR added devicetree and Kconfig for Ambiq Apollo4 Blue Plus SoC, needed for the apollo4p_blue_kxr_evb board.
Added devicetree, Kconfig and document for Ambiq Apollo4 Blue Plus KXR EVB board.

Samples Tested:
- samples/hello_world.
- samples/driver/counter/alarm.
- samples/driver/watchdog.

This should be merged after the PR (https://github.com/zephyrproject-rtos/hal_ambiq/pull/7) is merged.

carlescufi commented on west.yml:
```diff
@@ -142,7 +142,7 @@ manifest:
         - hal
     - name: hal_ambiq
-      revision: 0e15e4becd49c40cb054672d9525eccfe98b440f
+      revision: 174a4e3bbeedc7858fcaf8afaaf90b2e9d0de137
```
Please use `pull/7/head` as a revision as described here:
https://docs.zephyrproject.org/latest/develop/modules.html#process-for-submitting-changes-to-existing-modules

aaronyegx replied:
@carlescufi thanks for your comment. I updated the revision.

aaronyegx commented:
Hi @nashif @MaureenHelm @galak @ioannisg, could you help take a look on this PR? Appreciate your time.

MaureenHelm reviewed (dismissed):
Looks good. Just need to merge the HAL PR and update the west manifest accordingly

fkokosinski commented:
@aaronyegx I've merged the HAL PR. You should be able to bump the revision in `west.yml` now.

aaronyegx commented:
> Looks good. Just need to merge the HAL PR and update the west manifest accordingly

Hi @MaureenHelm, thanks for your review. The HAL update has been merged the west manifest has been update as well.

aaronyegx commented:
> @aaronyegx I've merged the HAL PR. You should be able to bump the revision in `west.yml` now.

Hi @fkokosinski, thanks for your time. I've update the "west.yml" and rebased the commits.

aaronyegx commented:
Hi @fkokosinski , I updated the west.yml to the latest hal_ambiq main branch commit hash exactly but it seems it reports build error in "Build with Clang/LLVM/ clang-build (native_posix)" item. Could you advise what does "exit code 4" mean and how can I fix it?

west.yml
=========
Manifest file 'west.yml' changed
=========
Error: Process completed with exit code 4.

aaronyegx committed:
board: arm: Add support for apollo4p_blue_kxr_evb board.

Add devicetree, Kconfig and doc for Ambiq Apollo4 Blue Plus KXR EVB board.

Signed-off-by: Aaron Ye <aye@ambiq.com>

aaronyegx committed:
soc: arm: Add support for Ambiq Apollo4 Blue Plus.

Added devicetree and Kconfig for Apollo4 Blue Plus SoC.
They are needed for the apollo4p_blue_kxr_evb board.

Signed-off-by: Aaron Ye <aye@ambiq.com>

aaronyegx committed:
manifest: Update hal_ambiq revision.

This change adds Apollo4 Blue Plus SoC support.

Signed-off-by: Aaron Ye <aye@ambiq.com>

aaronyegx commented:
Hi @fkokosinski , after rebasing the main branch sometimes it introduced new checking issues which did not relate with the changes of this PR. Should we keep rebasing the commits until passing all checks or we can ignore the no-related check failure to move forward the PR?

aaronyegx commented:
Hi @fkokosinski, @MaureenHelm , could you help take another look at this PR? Thanks very much.

pds72 commented:
Thankyou @fkokosinski, @MaureenHelm and @aaronyegx for approving this PR. This is the 1st Zephyr pull request and we are hoping to be an active member of the Zephyr community.
